src/mon/MonmapMonitor.cc

```diff
@@ -782,6 +782,35 @@ bool MonmapMonitor::prepare_command(MonOpRequestRef op)
     pending_map.set_addrvec(name, av);
     pending_map.last_changed = ceph_clock_now();
     propose = true;
+  } else if (prefix == "mon set-priority" ||
+             prefix == "mon set-weight") {
+    string name;
+    int64_t value;
+    const int64_t max_value = std::numeric_limits<uint16_t>::max();
+    if (!cmd_getval(g_ceph_context, cmdmap, "name", name) ||
+         (!cmd_getval(g_ceph_context, cmdmap, "priority", value) &&
+          !cmd_getval(g_ceph_context, cmdmap, "weight", value))) {
+      err = -EINVAL;
+      goto reply;
+    }
+    if (!pending_map.contains(name)) {
+      ss << "mon." << name << " does not exist";
+      err = -ENOENT;
+      goto reply;
+    }
+    if (value > max_value) {
```

Comment:
shall we enforce the minimum value also?

Reply:
It should have been done by the command para type sanity checking:

`"name=priority,type=CephInt,range=0"`

Reply:
ic. can you make it `range=0|65535` ? see https://tools.ietf.org/html/rfc2782#page-3

Reply:
Sure, added.